tools/mtmd/clip.cpp

```diff
@@ -3639,16 +3640,67 @@ struct llava_uhd {
         const bool has_slices    = original_size.width > slice_size || original_size.height > slice_size;
         const bool has_pinpoints = !ctx->model.hparams.image_res_candidates.empty();
 
-        if (!has_slices) {
-            // skip slicing logic
-            res.overview_size = clip_image_size{slice_size, slice_size};
-            res.refined_size  = clip_image_size{0, 0};
-            res.grid_size     = clip_image_size{0, 0};
+        if (clip_is_minicpmv(ctx)) {
```

Comment:
What's the change here compared to the initial version?

IMO I think this is more like a duplicated logic, see `clip_image_preprocess` on how this is used by minicpm-v